src/librustdoc/passes/check_code_block_syntax.rs

```diff
@@ -57,40 +58,56 @@ impl<'a, 'tcx> SyntaxChecker<'a, 'tcx> {
                     "Rust code block is empty"
                 };
 
-                let mut diag = self.cx.sess().struct_span_warn(sp, warning_message);
-
-                if code_block.syntax.is_none() && code_block.is_fenced {
-                    let sp = sp.from_inner(InnerSpan::new(0, 3));
-                    diag.span_suggestion(
-                        sp,
-                        "mark blocks that do not contain Rust code as text",
-                        String::from("```text"),
-                        Applicability::MachineApplicable,
-                    );
-                }
-
-                diag
+                // NOTE(poliorcetics): is 'item' always local here ? Or will this panic one day ?
+                let hir_id = self.cx.tcx.hir().local_def_id_to_hir_id(item.def_id.expect_local());
```

Comment:
It is *not* always local (e.g. if this is a re-export), I think we found why it wasn't a lint before. Fortunately, we don't need to check the syntax of other crates, so you can just return without doing anything if it's non-local.

See https://github.com/rust-lang/rust/issues/77230 for more discussion; it would also be good to add a test, since if the behavior changes it should be intentional.

Reply:
Could you add a test for the behavior?